src/libstd/error.rs

```diff
@@ -71,6 +74,14 @@ pub trait Error: Debug + Display {
     /// The lower-level cause of this error, if any.
     #[stable(feature = "rust1", since = "1.0.0")]
     fn cause(&self) -> Option<&Error> { None }
+
+    /// Get the `TypeId` of `self`
+    #[doc(hidden)]
+    #[unstable(feature = "core",
+               reason = "unclear whether to commit to this public implementation detail")]
+    fn type_id(&self) -> TypeId where Self: 'static {
```

Comment:
If you move the Reflect bound to a method-level where clause here, then we can use exclusively Any and avoid stabilizing Reflect.